src/angular-app/languageforge/lexicon/settings/configuration/field-unified-view.model.ts

```diff
@@ -96,16 +95,52 @@ export class ConfigurationFieldUnifiedViewModel {
     }
   }
 
-  static selectAllRow(setting: SettingsBase, settings: SettingsBase[], selectAll: SettingsBase): void {
+  static disableRequiredFields(fieldSettings: FieldSettings): boolean {
+    const disableFieldList = DisableFields.disableFieldList();
+    let disabled = false;
+    if (disableFieldList.includes(fieldSettings.fieldName)) {
+      disabled = true;
+    }
+    return disabled;
+  }
```

Comment:
name change to replace "disabled" with "required".
perhaps you could also simplify the function to a single `return` statement:

```suggestion
  static ensureRequiredFields(fieldSettings: FieldSettings): boolean {
    return RequiredFields.requiredFieldList().includes(fieldSettings.fieldName);
  }
```